---
title: "Wellcome Open Research R-script"
author: "R. Zach"
date: "19/12/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Loading Libraries
```{r message=FALSE, warning=FALSE, paged.print=FALSE}
library(dplyr)
library(tidyr)
library(ggplot2)
library(grid)
library(gridExtra)
library(cowplot)
library(gplots)
library(lattice)
library(ggthemes)
library(scales)
library(seqinr)
library(tidyverse)
library(ggforce)
```

## Definining Functions
### *import.multiple.csv.files*

**Arguments:**  
mypath: path to the folder containing files to be analysed  
genotype: genotype as character string

**Function description:**  
  Imports names of the .csv files contained in a given folder defined by mypath.
```{r}
import.multiple.csv.files <- function (mypath) {
  
  tmp.list.1 <- list.files(path = mypath, pattern = ".csv")
  tmp.list.2 <- list(length = length(tmp.list.1))
  for (i in 1:length(tmp.list.1)) {
    tmp.list.2[[i]] <- read.csv(paste(mypath, tmp.list.1[i], sep = "/"))
  }
  names(tmp.list.2) <- tmp.list.1
  for (a in 1:length(tmp.list.2)) {
    tmp.list.2[[a]] <- mutate(tmp.list.2[[a]], 
                              BiolRep = substring(names(tmp.list.2[a]), first = 7, last = 13),
                              TechRep = substring(names(tmp.list.2[a]), first = 15, last = 19),
                              strain = substring(names(tmp.list.2[a]), first = 1, last = 5),
                              pol = substring(names(tmp.list.2[a]), first = 21, last = 21),
                              strand = substring(names(tmp.list.2[a]), first = 23, last = 23))
  }
  return(tmp.list.2)
}
```

### *df.into.list*

**Arguments:**  
  data: data frame to be converted to the list object  
col: selected variable (column) as character string

**Function description:**  
  Takes a data frame and converts it into the list object based on variable of choice.
```{r}
df.into.list <- function (data, col) {
  data <- as.data.frame(data)
  tmp.list <- list()
  for (i in 1:length(unique(pull(data, col)))) {
    tmp.list[[i]] <- data[data[, col] == unique(pull(data, col))[i], ]
  }
  return(tmp.list)
}
```

### *sma*

**Arguments:**  
  data: data frame  
n: number of bins to each side of the 'center' bin  
col: column to bo smoothed as character  
newcol: name of the new column as character, smoothed data are store here

**Function description:**  
  Simple moving average function.

```{r}
sma <- function (data, n , col, newcol) {
  
  data <- as.data.frame(data)
  tmp.vector.input <- pull(data, col)
  tmp.vector.output <- vector()
  
  for (i in 1:length(tmp.vector.input)) {
    
    tmp.vector.output[i] <- mean(tmp.vector.input[i:(i+2*n)], na.rm = TRUE)
    
    if (i >= (length(tmp.vector.input) - 2*n)) {
      break
    }
  }
  data[, newcol] <- c(tmp.vector.input[1:n], tmp.vector.output, tmp.vector.input[(length(tmp.vector.input) - (n-1)):length(tmp.vector.input)])
  return(data)
}
```

### fun.diff

**Arguments:**  
  data: data frame  
col: column to be processed

**Function description:**  
  Calculates differential values (x(i) - x(i+1))

```{r}
fun.diff <- function (data, col) {
  tmp.data <- pull(data, col)
  tmp.vector <- as.vector(0)
  
  if ((length(tmp.data) >= 2)) {
    
    for (i in 2:length(tmp.data)) {
      tmp.vector[i] <- tmp.data[i] - tmp.data[i - 1]
      }
    data <- mutate(data, diff = tmp.vector)
  } else {
    data <- mutate(data, diff = tmp.vector)
  }
    
  return(data)}
```

### *peak.calling*

**Arguments:**  
  data: data frame  
bin_size: bin size (bp)  

**Function description:**  
  Assignes every distinct differential peak a number. Thus, every origin is labeled by unique identifier.

```{r}
peak.calling <- function (data, bin_size) {
  
  data <- mutate(data, peak_number = 1)
  
  for (i in 2:nrow(data)) {
    
    if (i != nrow(data) & data[i, "pos"] == data[i-1, "pos"] + bin_size) {
      data[i, "peak_number"] <- data[i-1, "peak_number"]
    } else if (i == nrow(data) & data[i, "pos"] == data[i-1, "pos"] + bin_size) {
      data[i, "peak_number"] <- data[i-1, "peak_number"]
    } else if (i == nrow(data) & data[i, "pos"] != data[i-1, "pos"] + bin_size) {
      data[i, "peak_number"] <- (data[i-1, "peak_number"] + 1)
    } else {
      data[i, "peak_number"] <- (data[i-1, "peak_number"] + 1)
    }
  }
  tmp.output <- mutate(data, peak_number = as.character(peak_number)) %>%
    arrange(chro, pos)
  
  return(tmp.output)
}
```

### *local.minima*

**Arguments:**  
  data: data frame  
n: number of bins within the frame of which two local minima are merged  

**Function description:**  
  Determines the local minima for every differential peak. Local minima are defined as positions (x), which satisfy the condition: diff(x - 1) > diff(x) < diff(x + 1). If two local minima are identified within the window of n bins (in our case 4), only the lower local minimum is considered.

```{r}
local.minima <- function (data, n, col) {
  
  data <- mutate(data, local_min = FALSE)
  tmp.peak <- unique(data$peak_number)
  tmp.output <- data.frame()
  
  for (p in 1:length(tmp.peak)) {
    
    tmp.subset.peak <- subset(data, peak_number == tmp.peak[p])
    tmp.subset.peak[1, "local_min"] <- TRUE
    tmp.subset.peak[nrow(tmp.subset.peak), "local_min"] <- TRUE
    
    if (nrow(tmp.subset.peak) >= (2 * n + 1)) {
      
      for (i in (n + 1):(nrow(tmp.subset.peak) - n)) {
        
        if (pull(tmp.subset.peak, col)[i] < pull(tmp.subset.peak, col)[i - 1] &
            pull(tmp.subset.peak, col)[i] < pull(tmp.subset.peak, col)[i + 1]) {
          
          tmp.subset.peak[i, "local_min"] <- TRUE
          
        }
      }
      
      if (sum(tmp.subset.peak$local_min) > 3) {
        
        for (a in (n + 1):(nrow(tmp.subset.peak) - n)) {
          
          tmp.subset.peak.window <- tmp.subset.peak[a:(a+(n-1)), ]
          
          if (sum(tmp.subset.peak.window$local_min, na.rm = TRUE) > 1) {
            
            
            tmp.close.min <- subset(tmp.subset.peak.window, local_min == TRUE)
            tmp.higher.min <- max(pull(tmp.close.min, col))
            tmp.higher.min.pos <- tmp.close.min[tmp.close.min[, col] == tmp.higher.min, ]$pos
            
            tmp.subset.peak <- mutate(tmp.subset.peak, local_min = ifelse(pos == tmp.higher.min.pos, FALSE, local_min))
          }
        }
      }
    }
    tmp.output <- bind_rows(tmp.output, tmp.subset.peak)
  }
  return(tmp.output)
}
```

### *sub.peak.calling*

**Arguments:**  
  data: data frame

**Function description:**  
  Since some identified differential peaks are multiple overlapping (unresolved) peaks, local minima are used for determination of so-called "sub-peaks". Further on, "sub-peaks" are treated as single unique differential peaks.

```{r}
sub.peak.calling <- function (data) {
  
  data <- mutate(data, sub_peak_number = 1)
  tmp.peak <- unique(data$peak_number)
  tmp.output <- data.frame()
  
  for (p in 1:length(tmp.peak)) {
    
    tmp.subset.peak <- subset(data, peak_number == tmp.peak[p])
    
    if ((sum(tmp.subset.peak$local_min) > 2)) {
      for (i in 2:nrow(tmp.subset.peak)) {
        if (tmp.subset.peak[i, ]$local_min == TRUE & i != nrow(tmp.subset.peak)) {
          tmp.subset.peak[i, "sub_peak_number"] <- tmp.subset.peak[i-1, "sub_peak_number"] +1
        } else if (tmp.subset.peak[i, ]$local_min == FALSE) {
          tmp.subset.peak[i, "sub_peak_number"] <- tmp.subset.peak[i-1, "sub_peak_number"]
        } else if (tmp.subset.peak[i, ]$local_min == TRUE & i == nrow(tmp.subset.peak)) {
          tmp.subset.peak[i, "sub_peak_number"] <- tmp.subset.peak[i-1, "sub_peak_number"]
        }
      }
      tmp.output <- bind_rows(tmp.output, tmp.subset.peak)
    } else {
      tmp.output <- bind_rows(tmp.output, tmp.subset.peak)
    }
  }
  return(tmp.output)
}
```

### *binning.fun*

**Arguments:**  
  data: data frame
binwidth: defines size of the bin
column: variable to be binned

**Function description:**  
  Bins the data. Size of the bin is defined.

```{r}
binning.fun <-function (data, binwidth, column) {
  
  tmp.output <- data.frame()
  
  for (i in seq(from = 1, to  = nrow(data), by = binwidth)) {
    
    tmp.data.1 <- data[i:(i + binwidth - 1), ]
    tmp.sum <- sum(tmp.data.1[, column], na.rm = TRUE)
    
    tmp.data.2 <- head(tmp.data.1, n = 1)
    tmp.data.2[, column] <- tmp.sum
    
    tmp.output <- bind_rows(tmp.output, tmp.data.2)
  }
  tmp.nrow <- nrow(tmp.output)
  tmp.output <- mutate(tmp.output, bin = 1:tmp.nrow)
  return(tmp.output)
}
```

### *binning.fun.nbin*

**Arguments:**  
  data: data frame
nbin: number of bins to be generated
column: variable to be binned

**Function description:**  
  Bins the data. Final number if bins is defined.

```{r}
binning.fun.nbin <-function (data, nbin, column) {

  tmp.data <- mutate(data, bin = 0, row = 1:n())
  binwidth <- nrow(data) / nbin
  
  if (nrow(data) > nbin & binwidth <= 1) {
    
    tmp.seq <- round(seq(from = 1, to  = nrow(data), by = binwidth))
    tmp.data <- mutate(tmp.data, bin = tmp.seq) %>%
      mutate(bin = ifelse(is.na(bin), nbin, bin))
    
  } else if (nrow(data) > nbin & binwidth > 1) {
    
  tmp.seq <- unique(round(seq(from = 1, to  = nrow(data), by = binwidth)))
  
    for (i in 1:length(tmp.seq)) {
    
    tmp.data <- mutate(tmp.data, bin = ifelse((row >= tmp.seq[i] & row < tmp.seq[i+1]), i, bin)) %>%
      mutate(bin = ifelse(is.na(bin), nbin, bin))
  }
  } 
return(tmp.data)
}
```

# Data analysis

## Importing polymerase usage sequencing data  
```{r}
path_data <- "/Users/r.z./Documents/Laboratory/Laboratory Diary/Pu-Seq libraries protocols/20201016 Pu-Seq PolD-OE A and B"

data_count <- 
import.multiple.csv.files(mypath = path_data) %>%
  do.call(what = bind_rows) %>%
  group_by(chro, pos, strain, pol, strand, BiolRep) %>%
  summarise(count = sum(count)) %>%
  ungroup() %>%
  df.into.list(col = "pol") %>%
  lapply(df.into.list, col = "strand") %>%
  lapply(lapply, group_by, strain, pol, strand, chro, pos, BiolRep) %>%
  lapply(lapply, summarise, count = sum(count)) %>%
  lapply(lapply, ungroup) %>%
  lapply(lapply, mutate, count_norm = count / sum(count, na.rm = TRUE)) %>%
  lapply(do.call, what = bind_rows) %>%
  do.call(what = bind_rows) %>%
  select(chro, pos, strain, pol, strand, count, count_norm, BiolRep) %>%
  mutate(PolD = ifelse(strain %in% c("00655", "00856", "01141"), "1xPolD",
                       ifelse(strain %in% c("0RZ57", "0RZ62", "0RZ68"), "2xPolD",
                              ifelse(strain %in% c("RZ112", "RZ116", "RZ118"), "4xPolD", "error"))))

print(head(data_count))
```

## Number of reads for every dataset
```{r}
print(unite(data = data_count, col = dataset, strain, pol, strand, BiolRep, sep = "-", remove = TRUE) %>%
        group_by(dataset) %>%
        summarise(count_sum = sum(count))
      )
```

## Calculating Polymerase track (PT)
PT = (f - r) / (f + r)
```{r}
data_track <- data_count %>%
  select(- c(count_norm, strain)) %>%
  df.into.list(col = "PolD") %>%
  lapply(df.into.list, col = "pol") %>%
  lapply(lapply, df.into.list, col = "BiolRep") %>%
  lapply(lapply, lapply, spread, key = strand, value = count) %>%
  lapply(lapply, lapply, mutate, pol_track = (f - r) / (f + r)) %>%
  lapply(lapply, lapply, select, - c(f, r)) %>%
  lapply(lapply, lapply, sma, col = "pol_track", newcol = "pol_track_smooth", n = 3) %>%
  lapply(do.call, what = bind_rows) %>%
  do.call(what = bind_rows)

data_track_mean <- data_track %>%
  group_by(chro, pos, pol, PolD) %>%
  summarise(pol_track = mean(pol_track)) %>%
  ungroup() %>%
  df.into.list(col = "PolD") %>%
  lapply(df.into.list, col = "pol") %>%
  lapply(lapply, sma, col = "pol_track", newcol = "pol_track_smooth", n = 3) %>%
  lapply(lapply, fun.diff, col = "pol_track_smooth") %>%
  lapply(lapply, mutate, diff = diff / 2) %>%
  lapply(do.call, what = bind_rows) %>%
  do.call(what = bind_rows)

print(head(data_track))
print(head(data_track_mean))
```

## Estimating origin efficiencies
```{r}
data_ori <- data_track_mean %>%
  df.into.list(col = "PolD") %>%
  lapply(mutate, diff = ifelse((pol == "a" | pol == "d"), - diff, diff)) %>%
  lapply(group_by, PolD, chro, pos) %>%
  lapply(summarise, diff = mean(diff)) %>%
  lapply(ungroup) %>%
  lapply(sma, n = 3, col = "diff", newcol = "diff_smooth") %>%
  lapply(df.into.list, col = "chro") %>%
  lapply(lapply, filter, diff_smooth > 0) %>%
  lapply(lapply, peak.calling, bin_size = 300) %>%
  lapply(lapply, local.minima, n = 4, col = "diff_smooth") %>%
  lapply(lapply, sub.peak.calling) %>%
  lapply(lapply, unite, col = "ori", peak_number, sub_peak_number, remove = FALSE) %>%
  lapply(lapply, filter, diff > 0) %>%
  lapply(lapply, group_by, PolD, chro, ori) %>%
  lapply(lapply, summarise, eff = sum(diff), pos_start = min(pos), pos_end = max(pos), max_diff = max(diff)) %>%
  lapply(lapply, mutate, pos_mid = pos_start + (ceiling(((pos_end - pos_start) / 300) / 2) * 300),
                         eff_norm = eff / max(eff)) %>%
  lapply(lapply, mutate, eff_quantile = ifelse(eff_norm <= .25, 25,
                                               ifelse(eff_norm > .25 & eff_norm <= .50, 50,
                                                      ifelse(eff_norm > .50 & eff_norm <= .75, 75, 100)))) %>%
  lapply(lapply, filter, max_diff > quantile(max_diff, probs = 0.3)) %>%
  lapply(do.call, what = bind_rows) %>%
  lapply(arrange, chro, pos_start) %>%
  lapply(mutate, ori = row_number()) %>%
  lapply(select, PolD, chro, ori, pos_start, pos_end, pos_mid, eff, eff_norm, eff_quantile) %>%
  do.call(what = bind_rows)
  
print(head(data_ori))
```

## Selecting regions around efficient origins
```{r}
data_initiation <- subset(data_ori, eff > 0.4 & PolD == "1xPolD") %>%
  df.into.list(col = "chro") %>%
  lapply(df.into.list, col = "ori") %>% 
  lapply(lapply, (function (data, range = 3000) {
    
    tmp.data <- subset(data_track_mean, chro == data$chro &
                                   pos >= data$pos_mid - range & pos <= data$pos_mid + range) %>%
      mutate(n_row = n(),
             ori = data$ori)
    
    return(tmp.data)
  })) %>%
  do.call(what = bind_rows) %>%
  filter(n_row == max(n_row)) %>%
  df.into.list(col = "PolD") %>%
  lapply(df.into.list, col = "chro") %>%
  lapply(lapply, df.into.list, col = "ori") %>%
  lapply(lapply, lapply, df.into.list, col = "pol") %>%
  lapply(lapply, lapply, lapply, mutate, pos_norm = seq(from = - 3000, to = 3000, by = 300)) %>%
  lapply(lapply, lapply, do.call, what = bind_rows) %>%
  lapply(lapply, do.call, what = bind_rows) %>%
  lapply(do.call, what = bind_rows) %>%
  do.call(what = bind_rows) %>%
  unite(col = "ID", chro, ori, sep = "_")

print(head(data_initiation))
```

## Number of selected regions (replication initiation)
```{r}
print(unite(data = subset(data_initiation, pos_norm == 0), col = dataset, pol, PolD, sep = "-", remove = TRUE) %>%
        group_by(dataset) %>%
        summarise(n = n()))
```

## Summary of selected regions around efficient origins
```{r}
data_initiation_summary <- data_initiation %>%
  group_by(PolD, pol, pos_norm) %>%
  summarise(pol_track_mean = mean(pol_track, na.rm = TRUE), pol_track_sd = sd(pol_track, na.rm = TRUE))

print(data_initiation_summary)
```

## Selecting termination zones
```{r}
data_termination <- subset(data_ori, PolD == "1xPolD" & eff > 0.2) %>%
  df.into.list(col = "chro") %>%
  lapply(function (data, eff_treshold = 0.4) {
     
    tmp.data.subset <- subset(data, eff > eff_treshold)
    tmp.ori <- tmp.data.subset$ori
    tmp.output <- data.frame()
    
    for (i in 1:(length(tmp.ori) - 1)) {
      
      ori.1 <- subset(data, ori == tmp.ori[i])$pos_end
      ori.2 <- subset(data, ori == tmp.ori[i+1])$pos_start
      
      tmp.region <- subset(data, pos_mid > ori.1 & pos_mid < ori.2)
      
      if (any(tmp.region$eff < eff_treshold)) {
        next
      } else {
        tmp.output <- bind_rows(tmp.output,
                                subset(data_track_mean, chro == unique(data$chro) & pos >= ori.1 & pos <= ori.2) %>%
                                mutate(region = i))
      }
    }
    return(tmp.output)
  }) %>%
  do.call(what = bind_rows) %>%
  unite(col = "ID", chro, region, remove = FALSE) %>%
  df.into.list(col = "PolD") %>%
  lapply(df.into.list, col = "pol") %>%
  lapply(lapply, df.into.list, col = "ID") %>%
  lapply(lapply, lapply, binning.fun.nbin, column = "pol_track", nbin = 21) %>%
  lapply(lapply, do.call, what = bind_rows) %>%
  lapply(do.call, what = bind_rows) %>%
  do.call(what = bind_rows) %>%
  filter(bin != 0) %>%
  group_by(ID, chro, PolD, pol, bin) %>%
  summarise(pol_track_binned = mean(pol_track)) %>%
  mutate(bin = - (max(bin-1)/2):(max(bin-1)/2)) %>%
  ungroup()

print(head(data_termination))
```

## Number of selected regions (termination zones)
```{r}
print(unite(data = subset(data_termination, bin == 0), col = dataset, pol, PolD, sep = "-", remove = TRUE) %>%
        group_by(dataset) %>%
        summarise(n = n()))
```

## Summary of selected termination zones
```{r}
data_termination_summary <- data_termination %>%
  group_by(PolD, pol, bin) %>%
  summarise(pol_track_binned_mean = mean(pol_track_binned, na.rm = TRUE), pol_track_binned_sd = sd(pol_track_binned, na.rm = TRUE))

print(head(data_termination_summary))
```

## GRAPHS

## Graph for Figure 1: Explanatory graph (rNMP counts and polymerase tracks)
ggsave("graph_expl.pdf", graph_expl, width = 2, height = 1.4, units = "in", scale = 2.2)
```{r}
graph_expl <- plot_grid(
  
  ggplot(subset(data_count, PolD == "1xPolD" & BiolRep == "repeat1" & chro == "chr2" & pos > 256000 & pos < 261000),
                     aes(x = pos/1000, y = count, fill = pol)) +
  geom_bar(stat = "identity", position = "identity", width = 0.3, colour = "#000000", size = 0.2, alpha = 0.5) +
  scale_y_continuous(expand = c(0,0)) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_fill_manual(values = alpha(c("#829c24", "#246e9c", "#d85d5d"), 0.8)) +
  facet_grid(strand ~ pol) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        panel.grid.major.y=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0.5,0.5,0.5,0.5), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 13),
        axis.text.x=element_blank(),
        axis.text.y=element_text(colour="#000000", size = 8),
        strip.text=element_text(size=13, colour = "#000000"),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
    ggplot(subset(data_track, PolD == "1xPolD" & BiolRep == "repeat1" & chro == "chr2" & pos > 256000 & pos < 261000)) +
  geom_line(aes(x = pos / 1000, y = pol_track, group = pol), size = 0.3, colour = "#000000") +
  geom_point(aes(x = pos/1000, y = pol_track, col = pol), shape = 21, size = 2, fill = "#FFFFFF") +
  scale_colour_manual(values = c("#829c24", "#246e9c", "#d85d5d")) +
  scale_y_continuous(expand = c(0,0), limits = c(-1, 1)) +
  scale_x_continuous(expand = c(0, 0)) +
  facet_grid(. ~ pol) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        panel.grid.major.y=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0.5,0.5,0.5,0.5), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_text(size = 13),
        axis.title.y = element_text(size = 13),
        axis.text.x=element_text(colour="#000000", size = 8),
        axis.text.y=element_text(colour="#000000", size = 8),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
  ncol = 1, nrow = 2, align = "hv", axis = "lr", rel_heights = c(1.1, 1))

graph_expl
```

## Graph for Figure 3: The right arm of chromosome III
ggsave("graph_chromosome3.pdf", Graph_chromosome3, width = 3, height = 1, units = "in", scale = 2)
```{r}
Graph_chromosome3 <-     ggplot(subset(data_track_mean,  chro == "chr3" & pos > 1137003)) +
  geom_line(aes(x = pos / 1000, y = pol_track, group = pol, colour = PolD), size = 0.3) +
  scale_colour_manual(values = c("#000000", "#FFB000", "#589B46")) +
  scale_y_continuous(expand = c(0,0), limits = c(-1, 1)) +
  scale_x_continuous(expand = c(0, 0)) +
  facet_grid(pol ~ .) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        panel.grid.major.y=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0.5,0.5,0.5,0.5), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_text(size = 13),
        axis.title.y = element_text(size = 13),
        axis.text.x=element_text(colour="#000000", size = 8),
        axis.text.y=element_text(colour="#000000", size = 8),
        strip.text=element_text(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top")

Graph_chromosome3
```




## Graph for Figure 4: Replication initiation heatmaps + summary
ggsave("graph_origins.pdf", graph_heatmap_origins, width = 1, height = 1, units = "in", scale = 3.3)
```{r}
graph_heatmap_origins <- plot_grid(
  
  ggplot(data = data_initiation, aes(x = pos_norm / 1000, y = ID, fill = pol, alpha = pol_track)) +
    geom_tile() +
    facet_grid(PolD ~ pol) +
    scale_fill_manual(values = c("#829c24", "#246e9c", "#d85d5d")) +
    scale_y_discrete(expand = c(0,0)) +
    scale_x_continuous(expand = c(0, 0)) +
    scale_alpha_continuous(range = c(0.1, 1)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(0.2, "lines"),
        plot.margin=unit(c(0.25,1,0,1), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 13),
        axis.text.x=element_blank(),
        axis.text.y=element_blank(),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
  ggplot() +
  geom_point(data = data_initiation_summary, aes(x = pos_norm / 1000, y = pol_track_mean, shape = PolD, colour = pol), size = 1.5, stroke = .2) +
  facet_grid(. ~ pol) +
  scale_shape_manual(values = c(21, 22, 24)) +
  scale_colour_manual(values = c("#829c24", "#246e9c", "#d85d5d")) +
  scale_fill_manual(values = alpha(c("#FFFFFF", "#FFFFFF", "#FFFFFF"), 0.5)) +
  scale_y_continuous(expand = c(0,0), limits = c(-1, 1), breaks = c(-0.5, 0, 0.5)) +
  scale_x_continuous(expand = c(0, 0), breaks = c(-2, 0, 2)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0,1,0.25,1), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.x=element_text(colour="#000000", size = 10),
        axis.text.y=element_text(colour="#000000", size = 10),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
    ncol = 1, nrow = 2, align = "hv", axis = "lr", rel_heights = c(1, 0.5), rel_widths = c(1, 1)
)

graph_heatmap_origins
```

## Graph for Figure 4: Replication termination heatmaps + summary
ggsave("graph_termination.pdf", graph_heatmap_termination, width = 1, height = 1, units = "in", scale = 3.3)
```{r}
graph_heatmap_termination <- plot_grid(
  
  ggplot(data = data_termination, aes(x = bin, y = ID, fill = pol, alpha = pol_track_binned)) +
    geom_tile() +
    facet_grid(PolD ~ pol) +
    scale_fill_manual(values = c("#829c24", "#246e9c", "#d85d5d")) +
    scale_y_discrete(expand = c(0,0)) +
    scale_x_continuous(expand = c(0, 0)) +
    scale_alpha_continuous(range = c(0.1, 1)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(0.2, "lines"),
        plot.margin=unit(c(0.25,1,0,1), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 13),
        axis.text.x=element_blank(),
        axis.text.y=element_blank(),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
  ggplot() +
  geom_point(data = data_termination_summary, aes(x = bin, y = pol_track_binned_mean, shape = PolD, colour = pol), size = 1.5, stroke = .2) +
  facet_grid(. ~ pol) +
  scale_shape_manual(values = c(21, 22, 24)) +
  scale_colour_manual(values = c("#829c24", "#246e9c", "#d85d5d")) +
  scale_fill_manual(values = alpha(c("#FFFFFF", "#FFFFFF", "#FFFFFF"), 0.5)) +
  scale_y_continuous(expand = c(0,0), limits = c(-1, 1), breaks = c(-0.5, 0, 0.5)) +
  scale_x_continuous(expand = c(0, 0), breaks = c(-8, -4, 0, 4, 8)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0,1,0.25,1), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.x=element_text(colour="#000000", size = 10),
        axis.text.y=element_text(colour="#000000", size = 10),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top"),
  
    ncol = 1, nrow = 2, align = "hv", axis = "lr", rel_heights = c(1, 0.5), rel_widths = c(1, 1)
)

graph_heatmap_termination
```

## Graph for Figure 4: Origin efficiency histograms
ggsave("graph_eff_histogram.pdf", graph_eff_histogram, width = 1, height = 0.4, units = "in", scale = 3)
```{r}
graph_eff_histogram <- ggplot(data = data_ori, aes(x = eff_norm * 100, col = PolD)) +
  geom_area(stat = "bin", binwidth = 1.5, position = "identity", alpha = 0.05, fill = NA) +
  scale_colour_manual(values = c("#000000", "#FFB000", "#589B46")) +
  scale_y_continuous(expand = c(0,0), limits = c(0, 70)) +
  scale_x_continuous(expand = c(0, 0)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0.25,0.25,0.25,0.25), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.x=element_text(colour="#000000", size = 8),
        axis.text.y=element_text(colour="#000000", size = 8),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top")

graph_eff_histogram
```

## Graph for Figure 4: origin efficiency quantiles
ggsave("graph_eff_quantiles.pdf", graph_eff_quantiles, width = 0.65, height = 0.4, units = "in", scale = 3)
```{r}
graph_eff_quantiles <- ggplot(data = data_ori, aes(x = as.factor(eff_quantile), fill = PolD)) +
  geom_bar(stat = "count", alpha = 0.5, position = position_dodge(), colour = "#000000", size = 0.3, width = 0.8) +
  scale_fill_manual(values = c("#000000", "#FFB000", "#589B46")) +
  scale_y_continuous(expand = c(0,0), limits = c(0, 560)) +
  scale_x_discrete(expand = c(0, 0)) +
    theme_tufte() +
    theme(axis.text=element_text(size= 8),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.spacing.y=unit(-0.05, "lines"),
        plot.margin=unit(c(0.25,0.25,0.25,0.25), "cm"),
        panel.border = element_rect(colour = "black", fill = NA, size = 0.6),
        panel.background = element_blank(),
        strip.background= element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.x=element_text(colour="#000000", size = 8),
        axis.text.y=element_text(colour="#000000", size = 8),
        strip.text=element_blank(),
        legend.key.size =  unit(0.9, "line"),
        axis.line = element_blank(),
        axis.ticks.x = element_line(colour = "#000000", size = 0.4),
        axis.ticks.y = element_line(colour = "#000000", size = 0.4),
        legend.text = element_text(size = 13),
        legend.title=element_blank(),
        legend.position = "none",
        legend.justification = "top")

graph_eff_quantiles
```





